### ISSUE TITLE
Throw exceptions instead of `exit(1)`

### ISSUE THREAD

**Issue by kevin-keraudren** [BioMedIA/irtk-refactored-deprecated#71]
`exit(1)` does not leave any chance to catch exceptions. It also kills interactive Python sessions.

The current code in IRTK looks like this:

https://github.com/BioMedIA/IRTK/blob/master/Modules/Image/src/irtkFileNIFTIToImage.cc#L173

```
    cerr << "irtkFileNIFTIToImage::ReadHeader: Number of dimensions > 5 (Number of dimensions = " << hdr.nim->dim[0] << ") \n";
    exit(1);
```

on the other hand the code in IRTK-LEGACY looks like this:

https://github.com/BioMedIA/IRTK/blob/master/Modules/Image/src/irtkFileNIFTIToImage.cc#L173

```
      stringstream msg;
      msg << "irtkFileNIFTIToImage::ReadHeader: Number of dimensions > 5 (Number of dimensions = " << hdr.nim->dim[0] << ") \n";
      cerr << msg.str();
      throw irtkException( msg.str(),
                           __FILE__,
                           __LINE__ );
```

Are you happy with `exit(1)`? Do you agree with the syntax and formating in IRTK-LEGACY or would rather have a different way of throwing exceptions?

Note that `irtkException`s are still there: https://github.com/BioMedIA/IRTK/blob/master/Modules/Common/include/irtkException.h

**Comment by kevin-keraudren** [comment]
These are the two relevant commits in IRTK-LEGACY:
https://github.com/BioMedIA/irtk-legacy/commit/6fac7725bd41ef64d593d7bfdedeef0dbcc0e8fa
https://github.com/BioMedIA/irtk-legacy/commit/c0ba69abc1a5de7d1fd6b8fea1f272c505f70e78

**Comment by ghisvail** [comment]
Indeed, this is a reminder of the fact that IRTK is not as ready as we like for consumption as a _public_ library. These `exit(1)` calls should disappear in favour of proper exception management such as the commits you pointed out. That being said, there are other calls to `exit(1)` elsewhere and I would welcome a PR that addresses all.
